Source/cmake/WebKitCommon.cmake

```diff
@@ -258,6 +261,67 @@ if (NOT HAS_RUN_WEBKIT_COMMON)
                     "${CMAKE_SOURCE_DIR}"
                     "${CMAKE_BINARY_DIR}"
         )
+    elseif (CMAKE_EXPORT_COMPILE_COMMANDS)
+        # Create a simple symlink in DeveloperTools/compile_commands.json so that
+        # it can be relied upon regardless of unified or non-unified builds.
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json
+            COMMAND ${CMAKE_COMMAND} -E create_symlink
+                    ../compile_commands.json
+                    ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            VERBATIM
+        )
+        add_custom_target(SymlinkCompileCommandsInDeveloperTools
+            ALL
+            DEPENDS ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+        )
+    endif ()
+
+    set(webkitbuild_dir ${CMAKE_SOURCE_DIR}/WebKitBuild)
+    cmake_path(IS_PREFIX webkitbuild_dir ${CMAKE_BINARY_DIR} NORMALIZE clangd_auto_setup_default)
```

Comment:
That would definitely be an option. This was my train of thought behind this default: "writing things in the source tree instead of the build tree is generally frowned upon, but we need to do it for clangd. However, if the user is building inside `WebKitBuild`, they're already doing a build inside their source tree, so it's not dirtying that much more.

However, I could use DEVELOPER_MODE, or even boolean OR of DEVELOPER_MODE or building inside `WebKitBuild`. What do you think makes the most sense?

Reply:
Understood. I have no preference. It's up to you.